table/block_based/block_based_table_builder.cc

```diff
@@ -742,13 +746,15 @@ void BlockBasedTableBuilder::Add(const Slice& key, const Slice& value) {
         r->index_builder->OnKeyAdded(key);
       }
     }
-    NotifyCollectTableCollectorsOnAdd(key, value, r->offset,
+    // TODO offset passed in is not accurate for parallel compression case
+    NotifyCollectTableCollectorsOnAdd(key, value, r->get_offset(),
```

Comment:
Maybe we can use `EstimatedFileSize` here.

Reply:
This place is kind of tricky. It is to give out a number to users. Some uses might be to estimate stats. I'm not sure whether we can make this number go back.